### PR TITLE
DOC: Examples added for float_format in to_csv documentation

### LINKED ISSUE #60363: DOC: Add examples for float_format in to_csv documentation
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_csv.html

### Documentation problem

The float_format parameter in to_csv is explained but lacks examples. Users might struggle to understand how to apply this parameter effectively without concrete examples in the documentation.

### Suggested fix for documentation

I suggest adding examples for float_format to make the documentation more beginner-friendly. Examples could include:

```
# Format floats to two decimal places
df.to_csv("example1.csv", float_format="%.2f")

# Use scientific notation
df.to_csv("example2.csv", float_format="{:.2e}".format)
```

### PR DESCRIPTION
**Pull request by Uvi-12**
Closes #60363

### CODE REVIEW

File: doc/source/getting_started/install.rst [hunk 1]
```diff
@@ -193,25 +193,25 @@ Visualization
 
 Installable with ``pip install "pandas[plot, output-formatting]"``.
 
-========================= ================== ================== =============================================================
```
Comment (mroeschke) [inline doc/source/getting_started/install.rst:196]: Could you undo all the changes in this file? They don't appear related to the linked issue

File: pandas/core/generic.py [hunk 2]
```diff
@@ -3878,6 +3878,12 @@ def to_csv(
         >>> import os  # doctest: +SKIP
         >>> os.makedirs("folder/subfolder", exist_ok=True)  # doctest: +SKIP
         >>> df.to_csv("folder/subfolder/out.csv")  # doctest: +SKIP
+
+        Format floating-point numbers using float_format:
+            - Format floats to two decimal places:
+              >>> df.to_csv("out1.csv", float_format="%.2f")  # doctest: +SKIP
+            - Use scientific notation:
+              >>> df.to_csv("out2.csv", float_format="{:.2e}".format)  # doctest: +SKIP
```
Comment (mroeschke) [inline pandas/core/generic.py:3886]:
```suggestion

        Format floats to two decimal places:
        
        >>> df.to_csv("out1.csv", float_format="%.2f")  # doctest: +SKIP
        
        Format floats using scientific notation:
        
        >>> df.to_csv("out2.csv", float_format="{:.2e}".format)  # doctest: +SKIP
```

**Review by mroeschke — CHANGES_REQUESTED** [review]

File: doc/source/getting_started/install.rst (hunk 1 above)
Comment (Uvi-12) [reply doc/source/getting_started/install.rst:196]: Some commits in this PR [#60457](https://github.com/pandas-dev/pandas/issues/60457)  were mistakenly added but are meant for issue https://github.com/pandas-dev/pandas/pull/60446.  I am new to open-source contributions and apologize for the confusion. 

Could you please guide me on how I can avoid this mistake in the future?

**Comment by Uvi-12** [comment]
@mroeschke The reviewed changes are corrected now.

**Comment by Uvi-12** [comment]
@mroeschke The code is changed accordingly to the reviewed changes, don't know why the tests are failing, can you please help me with it?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @Uvi-12

**Comment by Uvi-12** [comment]
@mroeschke Thank you for the reviews, that helped a lot.
